File: ms-ventas/src/main/java/com/smartlogix/msventas/model/Pedido.java
package com.smartlogix.msventas.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.OffsetDateTime;
import java.util.List;

@Entity
@Table(name = "pedido")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Pedido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPedido;

    private Long idCliente;

    private OffsetDateTime fechaCreacion;

    private Long montoTotal;

    private String estadoPedido;

    @OneToMany(mappedBy = "pedido", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonIgnore
    private List<DetallePedido> detalles;
}

File: frontend/src/features/pedidos/types/pedido.ts
export interface DetallePedido {
  idDetalle?: number;
  idPresentacion: number;
  cantidad: number;
  precioUnitarioSnapshot?: number;
}

export interface CrearPedidoDetallePayload {
  idPresentacion: number;
  cantidad: number;
}

export interface CrearPedidoPayload {
  idCliente: number;
  detalles: CrearPedidoDetallePayload[];
}

export interface Pedido {
  idPedido?: number;
  idCliente: number;
  fechaCreacion?: string;
  montoTotal: number;
  estadoPedido: string;
  detalles?: DetallePedido[];
}